model/patch/patch.go

```diff
@@ -301,6 +302,17 @@ func (p *Patch) SetParameters(parameters []Parameter) error {
 	)
 }
 
+func (p *Patch) SetDownstreamParameters(parameters []Parameter) error {
```

Comment:
Have you tested that `$addToSet` does what you expect? Would you consider adding unit tests for this function?

Reply:
added